src/main/cluster.ts

```diff
@@ -273,10 +274,12 @@ export class Cluster implements ClusterModel, ClusterState {
    */
   @action async init(port: number) {
     try {
+      this.initializing = true;
       this.contextHandler = new ContextHandler(this);
       this.kubeconfigManager = await KubeconfigManager.create(this, this.contextHandler, port);
       this.kubeProxyUrl = `http://localhost:${port}${apiKubePrefix}`;
       this.initialized = true;
+      this.initializing = false;
```

Comment:
set this in a `finally` block? As is, if `KubeconfigManager.create()` throws an error `this.initializing` will stay as `true`.

Reply:
That is probably desired (imo) since if it fails to be created then it isn't finished being `initialized`

Reply:
That doesn't seem to be the spirit of how `initializing` is used here. In `cluster-manager.ts` it is used to avoid calling `cluster.init()` while it is already running. If `this.initializing` remains as `true` after an error then `cluster.init()` will not get called again (according to the logic added in `cluster-manager.ts`). It's moot if it doesn't retry, or the `cluster` object gets recreated before trying again. I'm not sure what it does.

Reply:
I agree with @jim-docker , `finally` block would be better.

Reply:
Yes, good catch! Fixed.